tests/form_params_test.go

```diff
@@ -0,0 +1,385 @@
+		queryParamVal string
+	}
+	var tests = []struct {
+		name string
+		req  []byte
+		want want
+	}{
+		{
+			name: "Two Content-Length Headers",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" +
+				"Content-Length: 13\r\n" +
+				"Content-Length: 13\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+			want: want{
+				contentLen:    "13",
+				queryParamVal: "potato",
+			},
+		},
+		{
+			name: "List of same value in Content-Length Header",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" +
+				"Content-Length: 13, 13\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+			want: want{
+				contentLen:    "13",
+				queryParamVal: "potato",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		// Skipping to ensure GitHub checks are not failing. This test is made
+		// to fail intentionally to pinpoint behaviour we regard as inconsistent
+		t.Skip()
+		t.Run(test.name, func(t *testing.T) {
+			resp, err := requesttesting.MakeRequest(context.Background(), test.req, func(req *http.Request) {
+				if contentLen := req.Header["Content-Length"][0]; contentLen != test.want.contentLen {
+					t.Errorf(`req.Header["Content-Length"]: got %v, want %v`, contentLen, test.want.contentLen)
+				}
+				if val := req.FormValue("veggie"); val != test.want.queryParamVal {
+					t.Errorf("req.FormValue: want potato, got %s", val)
+				}
+			})
+			if err != nil {
+				t.Fatalf("MakeRequest(): got err %v, want nil", err)
+			}
+			if !!bytes.HasPrefix(resp, []byte(statusOKReq)) {
+				t.Errorf("response status: got %s, want %s", resp, statusOKReq)
+			}
+		})
+	}
+
+	postReq := []byte("POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" +
+		"Content-Length: 13\r\n" +
+		"Content-Length: 12\r\n" +
+		"\r\n" +
+		"veggie=potato\r\n" +
+		"\r\n")
+	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
+		if contentLen := req.Header["Content-Length"][0]; contentLen != "13" {
+			t.Error("MakeRequest(): request should have been rejected when list of values is given to Content-Length, rather inconsistent if you ask me.")
+		}
+	})
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(statusBadReq)) {
+		t.Errorf("response status: got %s, want %s", resp, statusBadReq)
+	}
+}
+
+// Test whether form parameters with Content-Type:
+// application/x-www-form-urlencoded that break percent-encoding will be
+// ignored
+func TestFormParametersBreakUrlEncoding(t *testing.T) {
+	postReq := []byte("POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: application/x-www-form-urlencoded\r\n" +
+		"Content-Length: 10\r\n" +
+		"\r\n" +
+		"veggie=%sc\r\n" +
```

Comment:
Add more parameters to check whether the parses gives up completely or parses other params.

Reply:
Resolved in [a523171](url)